### PR TITLE
Added releasemanagement module to angular and added camunda process engine

### PR DESCRIPTION
**Pull request by jandestiny**
- Created new folder "/camunda" for camunda process engine, which is in out-of-the-box state
- Added multiple components to angular project under /app/components/releasemanagement
- Added according angular services under /app/services/releasemanagement

This is still a work in progress. Commit history was unfortunately lost.

### CODE REVIEW

File: angular/src/app/components/release-management/release-detail/release-detail.component.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    this.isLoading = true;
+
+    this.camunda.getAllReleaseVariablesByInstanceId(this.release.processInstanceId)
+      .subscribe(
+        updatedVariables => {
+        this.release.variables = updatedVariables;
+        this.isLoading = false;
+        },
+        error => {
+          console.warn("Keine Variablen gefunden, Instanz womöglich gelöscht", error)
```
Comment (RammboNr5) [inline angular/src/app/components/release-management/release-detail/release-detail.component.ts:53]: isLoading stays true

File: angular/src/app/components/navigation/navigation.component.ts [hunk 2]
```diff
@@ -9,8 +9,8 @@ import { map, shareReplay } from 'rxjs/operators';
   styleUrls: ['./navigation.component.css']
 })
 export class NavigationComponent {
-  title: String = "ALLEGRO-Cockpit";
-  
+  title: String = "***REMOVED***-Cockpit";
```
[the remark this answers is not in the document]
Comment (jandestiny) [reply angular/src/app/components/navigation/navigation.component.ts]: fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

File: angular/src/app/components/release-management/create-new-release/create-new-release.component.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    })
+
+
+    //this.newReleaseForm.valueChanges.subscribe(console.log)
+
+  }
+
+  submitNewReleaseForm() {
+
+    if (this.newReleaseForm.valid) {
```
[the remark this answers is not in the document]
Comment (jandestiny) [reply angular/src/app/components/release-management/create-new-release/create-new-release.component.ts]: fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

File: angular/src/app/components/release-management/create-new-release/create-new-release.component.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+              })
+
+
+            this.router.navigateByUrl("/releases")
+
+          } else {
+
+            console.error("Lieferung konnte nicht erstellt werden.")
+
+            this.notify.open(`Lieferung koonnnte nicht erstellt werden, bitte prüfen Sie die eingegebenen Werte. ${response.statusText}`,
```
[the remark this answers is not in the document]
Comment (jandestiny) [reply angular/src/app/components/release-management/create-new-release/create-new-release.component.ts]: fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

File: angular/src/app/components/release-management/create-new-release/create-new-release.component.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      //convert input variables to ReleaseVariables
+      for (let key of this.startVariables.keys()) {
+        let updatedVariable = this.startVariables.get(key)
+        updatedVariable.value = this.newReleaseForm.value[key]
+        this.startVariables.set(key, updatedVariable)
+      }
+
+      this.camunda.createNewReleaseWithVariables(this.startVariables)
+        .subscribe(response => {
+          if (response.ok) {
```
[the remark this answers is not in the document]
Comment (jandestiny) [reply angular/src/app/components/release-management/create-new-release/create-new-release.component.ts]: fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

File: angular/src/app/components/release-management/edit-task/edit-task.component.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  selector: 'app-edit-task',
+  templateUrl: './edit-task.component.html',
+  styleUrls: ['./edit-task.component.css']
+})
+export class EditTaskComponent implements OnInit {
+
+  //Could be used later, if Camunda Embedded/External Forms are implemented.
+  // @ViewChild('taskFormHtml') taskFormHtml: ElementRef;
+
+  isLoading: boolean = true;
```
[the remark this answers is not in the document]
Comment (jandestiny) [reply angular/src/app/components/release-management/edit-task/edit-task.component.ts:17]: fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

File: angular/src/app/components/release-management/create-new-release/create-new-release.component.ts [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+              duration: 3000,
+              verticalPosition: 'bottom',
+              horizontalPosition: 'center'
+              })
+
+
+            this.router.navigateByUrl("/releases")
+
+          } else {
+
```
[the remark this answers is not in the document]
Comment (jandestiny) [reply angular/src/app/components/release-management/create-new-release/create-new-release.component.ts:97]: fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

File: angular/src/app/components/release-management/edit-task/edit-task.component.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  updateVariable(variableName: string, variableType: string, newValue: string) {
+    this.isLoading = true;
+
+    this.camunda.updateVariableByName(this.task.processInstanceId, variableName, newValue)
+      .subscribe(() => {
+        let newReleaseVariable: ReleaseVariable = { type: variableType, value: newValue, valueInfo: null };
+        this.task.variables.set(variableName, newReleaseVariable);
+        this.isLoading = false;
+      },
+        error => {
```
[the remark this answers is not in the document]
Comment (jandestiny) [reply angular/src/app/components/release-management/edit-task/edit-task.component.ts:50]: fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

File: angular/src/app/components/release-management/edit-task/edit-task.component.ts [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    this.camunda.submitTaskFormById(this.task.id, this.task.variables)
+      .subscribe(
+        success => {
+          //TODO: Add success notification
+          console.log("Erfolgreich abgeschlossen", success)
+          this.isLoading = false;
+          this.dialogReference.close()
+        },
+        error => {
+          //TODO: Add error notification
```
[the remark this answers is not in the document]
Comment (jandestiny) [reply angular/src/app/components/release-management/edit-task/edit-task.component.ts:69]: fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

File: angular/src/app/components/release-management/release-detail/release-detail.component.ts (hunk 1 above)
Comment (jandestiny) [reply angular/src/app/components/release-management/release-detail/release-detail.component.ts:54]: fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

File: angular/src/app/components/release-management/edit-task/edit-task.component.html [hunk 10]
```diff
@@ -0,0 +1,71 @@
+<div class="grid-container">
+  <mat-progress-bar *ngIf="isLoading" mode="indeterminate"></mat-progress-bar>
+  <h1 mat-dialog-title class="mat-h1">{{ task.name }}</h1>
+  <h3 mat-dialog-title class="mat-h3">Bitte die nachfolgenden Felder ausfüllen, um die Bearbeitung abschließen.</h3>
```
Comment (RammboNr5) [inline angular/src/app/components/release-management/edit-task/edit-task.component.html:4]: abzuschließen

File: angular/src/app/components/release-management/edit-task/edit-task.component.html [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+<mat-divider></mat-divider><br>
+
+<ng-container mat-dialog-title *ngIf="submitFormError">
+  <h3><mat-icon>warning</mat-icon>Fehler</h3>
+  <span>Bitte stellen Sie sicher, dass alle Felder ordnungsgemäß befüllt worden sind.</span><br>
+  <span>Leere Felder sind mit <mat-icon color="primary">info</mat-icon> markiert.</span>
```
Comment (RammboNr5) [inline angular/src/app/components/release-management/edit-task/edit-task.component.html:11]: gilt das für alle oder nur für Pflichtfelder? So wie es hier steht sind alle gemeint

File: angular/src/app/components/release-management/edit-task/edit-task.component.ts [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+
+  constructor(
+    public dialogReference: MatDialogRef<EditTaskComponent>,
+    @Inject(MAT_DIALOG_DATA) public task: CamundaTask,
+    private camunda: CamundaApiConnectorService,
+    public dialog: MatDialog
+  ) { }
+
+  ngOnInit(): void {
+    // this.camunda.getTaskFormByTaskId(this.task.id)
```
Comment (RammboNr5) [inline angular/src/app/components/release-management/edit-task/edit-task.component.ts:29]: remove commented code

File: angular/src/app/components/release-management/release-detail/release-detail.component.ts [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+    this.isLoading = true;
+
+    this.camunda.updateVariableByName(this.release.processInstanceId, variableName, newValue)
+      .subscribe(() => {
+        let newReleaseVariable: ReleaseVariable = { type: "String", value: newValue, valueInfo: null };
+        this.release.variables.set(variableName, newReleaseVariable);
+        this.isLoading = false;
+      },
+        error => {
+        console.error("Variable konnte nicht aktualisiert werden", error)
```
Comment (RammboNr5) [inline angular/src/app/components/release-management/release-detail/release-detail.component.ts:100]: this.isLoading = false

File: angular/src/app/components/release-management/release-management.component.ts [hunk 14]
```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit } from '@angular/core';
+import { VersionDataMockService } from '../../services/releasemanagement/version-data-mock.service'
+// import { CamundaApiMockService } from '../../services/releasemanagement/camunda-api-mock.service'
```
Comment (RammboNr5) [inline angular/src/app/components/release-management/release-management.component.ts:3]: delete this

File: angular/src/app/services/releasemanagement/camunda-api-connector.service.ts [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+  }
+
+  private postTaskForm(taskId: string, request_body: Object): Observable<HttpResponse<Object>> {
+    let url = this.baseUrl + "/task/" + taskId + "/submit-form";
+    return this.http.post(url, request_body, { observe: "response" })
+  }
+
+    //Task methods
+
+  // private getCurrentTasks(): Observable<CamundaTask[]> {
```
Comment (RammboNr5) [inline angular/src/app/services/releasemanagement/camunda-api-connector.service.ts:129]: delete commented code

File: camunda/README.md [hunk 16]
```diff
@@ -0,0 +1,56 @@
+
+# Release-Management-System: Workflow Engine
+
+Dieser Ordner enthält eine Instanz von Camunda Platform 7, die im Release-Management-System des Informations-Cockpits genutzt wird.
```
Comment (RammboNr5) [inline camunda/README.md:4]: IT-Cockpit

File: angular/src/app/components/release-management/edit-task/edit-task.component.html (hunk 10 above)
Comment (jandestiny) [reply angular/src/app/components/release-management/edit-task/edit-task.component.html:4]: done

File: angular/src/app/components/release-management/edit-task/edit-task.component.html (hunk 11 above)
Comment (jandestiny) [reply angular/src/app/components/release-management/edit-task/edit-task.component.html:11]: Nur Pflichtfelder, done

File: angular/src/app/components/release-management/edit-task/edit-task.component.ts (hunk 12 above)
Comment (jandestiny) [reply angular/src/app/components/release-management/edit-task/edit-task.component.ts:29]: done

File: angular/src/app/components/release-management/release-detail/release-detail.component.ts (hunk 13 above)
Comment (jandestiny) [reply angular/src/app/components/release-management/release-detail/release-detail.component.ts:100]: done

File: angular/src/app/services/releasemanagement/camunda-api-connector.service.ts (hunk 15 above)
Comment (jandestiny) [reply angular/src/app/services/releasemanagement/camunda-api-connector.service.ts:129]: done

**Pull request merged**